### PR TITLE
Fix docstrings for `gensim.models.rpmodel`

### PR DESCRIPTION
**Pull request by jazzmuesli**

### CODE REVIEW

File: gensim/models/rpmodel.py [hunk 1]
```diff
@@ -5,41 +5,54 @@
 # Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
 
 
-import logging
+"""
+Objects of this class allow building and maintaining a model for Random Projections
+(also known as Random Indexing). 
 
-import numpy as np
+For theoretical background on RP, see:   Kanerva et al.: "Random indexing of text samples for Latent Semantic Analysis."
```
Comment (menshikh-iv) [inline gensim/models/rpmodel.py]: Need to add concrete link to paper, something like this - https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/parsing/porter.py#L6

File: gensim/models/rpmodel.py [hunk 2]
```diff
@@ -5,41 +5,54 @@
...
 
-from gensim import interfaces, matutils, utils
+    The main methods are:
```
Comment (menshikh-iv) [inline gensim/models/rpmodel.py]: Better to add example here (sometimes code tell us more than text)

File: gensim/models/rpmodel.py [hunk 3]
```diff
@@ -5,41 +5,54 @@
...
 
 
-class RpModel(interfaces.TransformationABC):
-    """
-    Objects of this class allow building and maintaining a model for Random Projections
-    (also known as Random Indexing). For theoretical background on RP, see:
+Examples:
+---------
+>>> from gensim.models import rpmmodel
+>>> rp = RpModel(corpus)
```
Comment (menshikh-iv) [inline gensim/models/rpmodel.py]: An example isn't executable (i.e. corpus isn't defined), "executable" means that I can copy-paste it to console and this runs successfully.

File: gensim/models/rpmodel.py [hunk 4]
```diff
@@ -96,5 +125,12 @@ def __getitem__(self, bow):
         ]
 
     def __setstate__(self, state):
+        """
+        Sets the internal state and updates freshly_loaded to True. Called when unpicked.
+
+        Parameters
+        ----------
+        state : state of the class
```
Comment (anotherbugmaster) [inline gensim/models/rpmodel.py]:
```
state : dict
    State of the class
```

File: gensim/models/rpmodel.py [hunk 5]
```diff
@@ -5,41 +5,54 @@
...
     def __init__(self, corpus, id2word=None, num_topics=300):
         """
         `id2word` is a mapping from word ids (integers) to words (strings). It is
         used to determine the vocabulary size, as well as for debugging and topic
         printing. If not set, it will be determined from the corpus.
+
+
+        Parameters
+        ----------
+        corpus : interfaces.CorpusABC
```
Comment (menshikh-iv) [inline gensim/models/rpmodel.py]: please use 
```
:class:`~gensim.interfaces.CorpusABC`
```

when you reference a class from anywhere (gensim or something else)

Comment (anotherbugmaster) [inline gensim/models/rpmodel.py]: Add ``:class:``

File: gensim/models/rpmodel.py [hunk 6]
```diff
@@ -5,41 +5,54 @@
...
+        id2word : dict of int tostring
```
Comment (anotherbugmaster) [inline gensim/models/rpmodel.py]: `dict of (int, string)`

File: gensim/models/rpmodel.py [hunk 7]
```diff
@@ -75,6 +99,16 @@ def initialize(self, corpus):
     def __getitem__(self, bow):
         """
         Return RP representation of the input vector and/or corpus.
+
+        Parameters
+        ----------
+        bow : :class:`~interfaces.CorpusABC` (iterable of documents) or list of (int, int).
+
+        Examples:
```
Comment (anotherbugmaster) [inline gensim/models/rpmodel.py]: Redundant `:`

File: gensim/models/rpmodel.py [hunk 8]
```diff
@@ -75,6 +99,16 @@ def initialize(self, corpus):
...
+        -------------
+        >>> rp = RpModel(corpus)
```
Comment (anotherbugmaster) [inline gensim/models/rpmodel.py]: `corpus` is undefined

File: gensim/models/rpmodel.py [hunk 9]
```diff
@@ -5,41 +5,59 @@
...
 
-logger = logging.getLogger('gensim.models.rpmodel')
+Model persistency is achieved via its load/save methods.
 
 
-class RpModel(interfaces.TransformationABC):
-    """
-    Objects of this class allow building and maintaining a model for Random Projections
-    (also known as Random Indexing). For theoretical background on RP, see:
+Examples:
```
Comment (anotherbugmaster) [inline gensim/models/rpmodel.py]: Redundant `:`

File: gensim/models/rpmodel.py [hunk 10]
```diff
@@ -55,8 +60,13 @@ def __init__(self, corpus, id2word=None, num_topics=300):
         Parameters
         ----------
         corpus : :class:`~gensim.interfaces.CorpusABC` 
+            Iterable of documents
+
         id2word : dict of (int, string)
+            Mapping from word ids (integers) to words (strings)
```
Comment (anotherbugmaster) [inline gensim/models/rpmodel.py]: Redundant type definition

**Comment by markroxor** [comment]
@menshikh-iv is this PR meant to convert __all__ the documentation available in the gensim repo?

**Comment by menshikh-iv** [comment]
@markroxor no, this is only for one file (incorrect PR title)

**Comment by menshikh-iv** [comment]
@jazzmuesli thank you for PR, I'll look into on next week (sorry for waiting).

**Comment by menshikh-iv** [comment]
I add several fixes.

Thank you @jazzmuesli, congratz with your first contribution :1st_place_medal:

**Pull request merged**